lib/flowViewport/sceneIndex/fvpPathInterface.h

```diff
@@ -35,7 +35,12 @@ namespace FVP_NS_DEF {
 struct PrimSelection
 {
     PXR_NS::SdfPath primPath;
-    PXR_NS::HdDataSourceBaseHandle selectionDataSource;
+    std::optional<int> instanceIndex;
+
+    inline bool operator==(const PrimSelection &rhs) const {
+        return primPath == rhs.primPath
+            && instanceIndex == rhs.instanceIndex;
+    }
```

Comment:
Fundamental design change of the pull request : `PrimSelection` objects, returned by the different data producers, no longer return selection data sources directly, as data sources can't be efficiently compared. This posed problems when wanting to remove only specific selections (such as instance selections) on a prim in `Fvp::Selection::Remove`. 

`PrimSelection`s now only contain the core data required to create the selection data sources, which are now created by the `Fvp::Selection`. The information stored in `PrimSelection` is based off the `HdSelectionSchema` class, which currently only specifies full prim and instance selections.